src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -394,163 +310,85 @@ HOSTFXR_UTILITY::ParseHostfxrArguments(
     return hr;
 }
 
-HRESULT
+std::optional<fs::path>
 HOSTFXR_UTILITY::GetAbsolutePathToDotnet(
-    _Inout_ STRU* pStruAbsolutePathToDotnet
+     fs::path & requestedPath
 )
 {
-    HRESULT             hr = S_OK;
-
     //
     // If we are given an absolute path to dotnet.exe, we are done
     //
-    if (UTILITY::CheckIfFileExists(pStruAbsolutePathToDotnet->QueryStr()))
+    if (exists(requestedPath))
     {
-        goto Finished;
+        return std::make_optional(requestedPath);
     }
 
-    //
-    // If the path was C:\Program Files\dotnet\dotnet
-    // We need to try appending .exe and check if the file exists too.
-    //
-    if (FAILED(hr = pStruAbsolutePathToDotnet->Append(L".exe")))
+    const auto pathWithExe = requestedPath.append(L".exe");
+    if (exists(pathWithExe))
     {
-        goto Finished;
-    }
-
-    if (UTILITY::CheckIfFileExists(pStruAbsolutePathToDotnet->QueryStr()))
-    {
-        goto Finished;
+        return std::make_optional(pathWithExe);
     }
 
     // At this point, we are calling where.exe to find dotnet.
     // If we encounter any failures, try getting dotnet.exe from the
     // backup location.
-    if (!InvokeWhereToFindDotnet(pStruAbsolutePathToDotnet))
+    // Only do it if no path is specified
+    if (!requestedPath.has_parent_path())
     {
-        hr = GetAbsolutePathToDotnetFromProgramFiles(pStruAbsolutePathToDotnet);
+        return std::nullopt;
     }
 
-Finished:
+    const auto dotnetViaWhere = InvokeWhereToFindDotnet();
+    if (dotnetViaWhere.has_value())
+    {
+        return dotnetViaWhere;
+    }
 
-    return hr;
+    return GetAbsolutePathToDotnetFromProgramFiles();
 }
 
-HRESULT
+std::optional<fs::path>
 HOSTFXR_UTILITY::GetAbsolutePathToHostFxr(
-    STRU* pStruAbsolutePathToDotnet,
-    HANDLE hEventLog,
-    STRU* pStruAbsolutePathToHostfxr
+    fs::path & dotnetPath,
+    HANDLE hEventLog
```

Comment:
For the future: remove hEventLog from hostfxr_utility and log to event log outside of this function.